### PR TITLE
cli: validate services when unattached (SC-805)

### PR DESCRIPTION
**Pull request by renanrodrigo**
When trying to enable/disable, check if the services are valid and
present messages accordingly.

Added a new message name/content for the case when the user provides valid and invalid services to the CLI at the same time.

Fixes: #1976 

## Desired commit type
 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist
 - [x] I have updated or added any unit tests accordingly
 - [x] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

File: uaclient/cli.py [hunk 1]
```diff
@@ -206,20 +206,38 @@ def new_f(cmd_args, *args, **kwargs):
     return new_f
 
 
-def assert_attached(unattached_msg_tmpl=None):
-    """Decorator asserting attached config.
-
-    :param unattached_msg_tmpl: Optional msg template to format if raising an
-        UnattachedError
-    """
+def assert_attached():
```
Comment (CalvoM) [inline uaclient/cli.py:209]: Just a question, why did you move the msg from here?

Comment (renanrodrigo) [reply uaclient/cli.py:209]: Originally, the message was designed in a way that there is a generic "Unattached Error", but some commands could define their own messages.

I changed it because now there is logic to determine when to use different messages, even for a single command like `enable`. As we need different messages and specific logic, even passing a list of messages would not do, and getting the command from the args is easier than passing information through the decorator imho.

File: uaclient/cli.py [hunk 2]
```diff
@@ -206,20 +206,38 @@ def new_f(cmd_args, *args, **kwargs):
...
+                    elif entitlements_found:
+                        msg = messages.VALID_SERVICE_FAILURE_UNATTACHED.format(
+                            valid_service=", ".join(entitlements_found)
+                        )
+                    else:
+                        msg = messages.INVALID_SERVICE_OP_FAILURE.format(
+                            operation=command,
+                            invalid_service=", ".join(entitlements_not_found),
+                            service_msg="See https://ubuntu.com/advantage",
+                        )
```
[the remark this answers is not in the document]
Comment (renanrodrigo) [reply uaclient/cli.py]: @lucasmoura done - it's setting up a function that takes args to ensure the flexibility for any other command

File: uaclient/cli.py [hunk 3]
```diff
@@ -908,9 +906,40 @@ def action_config_unset(args, *, cfg, **kwargs):
     return 0
 
 
+def _create_enable_disable_unattached_msg(args):
```
Comment (lucasmoura) [inline uaclient/cli.py:909]: I think the function should receive the command and services as an argument instead, just so we can better decouple it from the cli args

**Review by lucasmoura — APPROVED** [review]
Just a minor nit, but LGTM

File: uaclient/cli.py [hunk 4]
```diff
@@ -204,20 +204,20 @@ def new_f(cmd_args, *args, **kwargs):
...
     def wrapper(f):
         @wraps(f)
         def new_f(args, cfg, **kwargs):
             if not cfg.is_attached:
-                if unattached_msg_tmpl:
-                    names = getattr(args, "service", "None")
-                    msg = unattached_msg_tmpl.format(name=", ".join(names))
+                if msg_function:
+                    command = getattr(args, "command", "None")
+                    service_names = getattr(args, "service", "None")
```
Comment (lucasmoura) [inline uaclient/cli.py:219]: Any reason not to use empty strings as default values ?

Comment (renanrodrigo) [reply uaclient/cli.py:219]: I just followed what was already there hahaha
But no, we don't even expect those to be absent at all.
Will change for consistency sake

Comment (renanrodrigo) [reply uaclient/cli.py]: done

**Review by lucasmoura — APPROVED** [review]
